test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -0,0 +1,455 @@
+          () {
+        testSuggestor(
+          expectedPatchCount: 0,
+          input: r'''
+      @Factory()
+      UiFactory<FooProps> Foo =
+          // ignore: undefined_identifier
+          $Foo;
+
+      @Props()
+      class _$FooProps extends ADifferentPropsClass with AMixin, AnotherMixin {
+        String foo;
+        int bar;
+      }
+
+      @State()
+      class _$FooState extends ADifferentStateClass with AStateMixin, AnotherStateMixin {
+        String foo;
+        int bar;
+      }
+
+      @Component2()
+      class FooComponent extends UiStatefulComponent2<FooProps, FooState> {
+        @override
+        render() {
+          return Dom.ul()(
+            Dom.li()('Foo: ', props.foo),
+            Dom.li()('Bar: ', props.bar),
+          );
+        }
+      }
+    ''',
+        );
+
+        expect(propsAndStateClassNamesConvertedToNewBoilerplate, isEmpty);
+      });
+    });
+
+    group('operates when the classes are advanced', () {
+      group('and there are both a props and a state class', () {
+        test('and the classes extend from a custom class', () {
+          testSuggestor(
+            expectedPatchCount: 12,
+            input: r'''
+        @Factory()
+        UiFactory<FooProps> Foo =
+            // ignore: undefined_identifier
+            $Foo;
+
+        @Props()
+        class _$FooProps extends ADifferentPropsClass {
+          String foo;
+          int bar;
+        }
+
+        @State()
+        class _$FooState extends ADifferentStateClass {
+          String foo;
+          int bar;
+        }
+
+        @Component2()
+        class FooComponent extends UiStatefulComponent2<FooProps, FooState> {
+          @override
+          render() {
+            return Dom.ul()(
+              Dom.li()('Foo: ', props.foo),
+              Dom.li()('Bar: ', props.bar),
+            );
+          }
+        }
+      ''',
+            expectedOutput: r'''
+      @Factory()
+        UiFactory<FooProps> Foo =
+            // ignore: undefined_identifier
+            $Foo;
+
+        @Props()
+        mixin FooPropsMixin on UiProps {
+          String foo;
+          int bar;
+        }
+
+        class FooProps = UiProps with ADifferentPropsClassMixin, FooPropsMixin;
+
+        @State()
+        mixin FooStateMixin on UiState {
+          String foo;
+          int bar;
+        }
+
+        class FooState = UiState with ADifferentStateClassMixin, FooStateMixin;
+
+        @Component2()
+        class FooComponent extends UiStatefulComponent2<FooProps, FooState> {
+          @override
+          render() {
+            return Dom.ul()(
+              Dom.li()('Foo: ', props.foo),
+              Dom.li()('Bar: ', props.bar),
+            );
+          }
+        }
+      ''',
+          );
+
+          expect(propsAndStateClassNamesConvertedToNewBoilerplate, {
+            'FooProps': 'FooPropsMixin',
+            'FooState': 'FooStateMixin',
+          });
```

Comment:
Should this be...

```diff
 expect(propsAndStateClassNamesConvertedToNewBoilerplate, {
   'FooProps': 'FooPropsMixin',
   'FooState': 'FooStateMixin',
+  'ADifferentStateClass': 'ADifferentStateClassMixin',
 });
```

Or is that piece of the map added in a portion of logic that is not exercised in this test?

Reply:
Yep, it was excluded because that class didn't get converted in this test case, but we're assuming it would be